api/v1alpha3/azuremachinetemplate_conversion.go

```diff
@@ -44,6 +45,10 @@ func (src *AzureMachineTemplate) ConvertTo(dstRaw conversion.Hub) error { // nol
 		}
 	}
 
+	if restored.Spec.Template.Spec.Image.SharedGallery != nil {
+		dst.Spec.Template.Spec.Image.SharedGallery = restored.Spec.Template.Spec.Image.SharedGallery
```

Comment:
@whites11 I made a mistake in my earlier review comments. We should only be setting the new fields here. This should work, but it'd be better if you can explicitly set the new fields only. eg `dst.Spec.Template.Spec.Image.SharedGallery.Publisher = restored.Spec.Template.Spec.Image.SharedGallery.Publisher` and so on.. Sorry about the confusion

Reply:
no worries at all, changed